webrender/src/device/gfx/device.rs

```diff
@@ -675,15 +596,68 @@ impl<B: hal::Backend> Device<B> {
             frame_images,
             viewport,
             _frame_count,
-        ) = Device::init_swapchain_resources(
-            &self.device,
-            &mut self.heaps,
-            &self.adapter,
-            self.surface.as_mut().unwrap(),
-            window_size,
-        );
+        ) = if self.surface.is_some() {
```

Comment:
I think `if let Some(surface) = self.surface` would be better

Reply:
Thank you! I just updated the PR with this change.